### ISSUE TITLE
JavaScript driver provided in npm sometimes fails when executing in strict mode

### ISSUE THREAD

**Issue by niieani** [rethinkdb/rethinkdb#5198]
I'm using rethinkdb 2.2.0 JS driver from `npm`. At times, I'm getting this during execution in strict mode:
```
Unhandled promise rejection TypeError: 'caller', 'callee', and 'arguments' properties may not be accessed on strict mode functions or the arguments objects for calls to them
```
CoffeeScript seems to have resolved the issue here some time ago (https://github.com/jashkenas/coffeescript/issues/1547), so the only reason why the code might still be non-compliant is if the person who is submitting the compiled version to `npm` is using an outdated version of CoffeeScript to compile the code. 

Would you kindly fix the `npm` version? Thanks!

**Comment by danielmewes** [comment]
Thanks for the heads up @niieani 

We'll look into it (I think I used 1.7.0 when generating the code for the NPM package).

We should also bump the coffeescript version that we download via `mk/support/pkg/coffee-script.sh` when fetching it.

@AtnNn Did you install `/usr/local/bin/coffee` on newton? Can you think of anything we need to keep in mind when updating it?

**Comment by AtnNn** [comment]
@danielmewes No, that was @neumino. I believe it was installed by doing `npm install -g coffee-script`.

**Comment by danielmewes** [comment]
A patch to bump the supported coffeescript version is now in internal review 3395.

**Comment by danielmewes** [comment]
Fixed in `v2.2.x` 55b8dc6 and `next` 3ec0642.
We'll release an updated driver in the next days with RethinkDB 2.2.2.
